src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -50,7 +54,13 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
                                     (void **) &xpmem_segids, MPL_MEM_SHM);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
-    mpi_errno = MPIDU_shm_seg_commit(&shm_seg,
+
+    mpi_errno = MPIDU_shm_seg_alloc(num_local * num_local * sizeof(OPA_int_t),
+                                    (void **) &xpmem_counter, MPL_MEM_SHM);
+    if (mpi_errno)
+        MPIR_ERR_POP(mpi_errno);
+
```

Comment:
`xpmem_counter` has to be shared until finalize. However, the `xpmem_segids` part is only needed in init, and should be freed here. Actually, you can just initialize the counter on P0 and use `XPMEM` to share with others.